Apps/Database/src/Delegates/DBDrugLookupDelegate.cs

```diff
@@ -52,12 +51,15 @@ public List<DrugProduct> GetDrugProductsByDIN(List<string> drugIdentifiers)
             this.logger.LogDebug($"Getting list of drug products from DB");
             this.logger.LogTrace($"Identifiers {JsonSerializer.Serialize(drugIdentifiers)}");
             List<string> uniqueDrugIdentifers = drugIdentifiers.Distinct().ToList();
-            List<DrugProduct> retVal = this.dbContext.DrugProduct.Where(dp => uniqueDrugIdentifers.Contains(dp.DrugIdentificationNumber))
-                                        .Include(c => c.Company)
-                                        .Include(a => a.ActiveIngredient)
-                                        .Include(f => f.Form)
-                        .ToList();
-
+            List<DrugProduct> retVal = this.dbContext.DrugProduct
+                                            .Include(c => c.Company)
+                                            .Include(a => a.ActiveIngredient)
+                                            .Include(f => f.Form)
+                                            .Where(dp => uniqueDrugIdentifers.Contains(dp.DrugIdentificationNumber))
+                                            .ToList()
```

Comment:
Do we need this .ToList() here?

Reply:
Yes, the query doesn't translate to a native SQL query, so the the first tolist performs the actual query and then the groupby and first occur on the client (our server).